### ISSUE TITLE
NameError: name 'logger' is not defined

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#24]
I get this error:

```
...snip...
  File "/u/int5sys/opt/mercurial-1.5/lib/python2.6/site-packages/mercurial/demandimport.py", line 106, in _demandimport
    mod = _origimport(name, globals, locals)
  File "/u/ehaszla/x/lib/python2.6/site-packages/keyring-0.2-py2.6.egg/keyring/core.py", line 122, in <module>
    init_backend()
  File "/u/ehaszla/x/lib/python2.6/site-packages/keyring-0.2-py2.6.egg/keyring/core.py", line 40, in init_backend
    keyring_impl = load_config()
  File "/u/ehaszla/x/lib/python2.6/site-packages/keyring-0.2-py2.6.egg/keyring/core.py", line 116, in load_config
    logger.warning("Keyring Config file does not write correctly.\n" + \
NameError: name 'logger' is not defined
```

when there is a problem with the keyringrc.cfg file.  It looks like the "logger" that core.py is trying to use is defined in **init**.py, but that isn't being loaded for some reason.  Perhaps this is a problem with how mercurial is loading the keyring module?

I installed the keyring and mercurial_keyring modules into a non-default location with "easy_install --prefix /u/ehaszla/x ..." b/c I do not have root access on the machine.

Also, even if using logger worked, the error message it would have displaying is a bit off; it should say something like:

```
  The keyring config file (~/keyringrc.cfg) is not properly formatted.
```

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/24
- Originally reported by: [erh](http://bitbucket.org/erh)
- Originally created at: 2010-06-23T00:01:27.870

**Comment by jaraco** [comment]
That's quite strange indeed. I could not reproduce it so far

---

Original comment by: Tarek Ziadé

**Comment by jaraco** [comment]
Maybe related to Issue #34

---

Original comment by: Tarek Ziadé

**Comment by jaraco** [comment]
According to the version in the traceback, you're running 0.2. I've updated to that version in the repo and the issue is quite straightforward - the logger import was missing. This issue can definitely be closed as of 0.3 - the import was added in 2009.

---

Original comment by: [getoffmalawn](http://bitbucket.org/getoffmalawn)

**Comment by jaraco** [comment]
This has actually been fixed in [https://bitbucket.org/kang/python-keyring-lib/changeset/81](81), 3 years ago.

---

Original comment by: Jonathan Ballet
